apps/s_client.c

```diff
@@ -3361,12 +3361,54 @@ int s_client_main(int argc, char **argv)
     return ret;
 }
 
+#ifndef OPENSSL_NO_EC
+static char *ec_curve_name(EVP_PKEY *pkey)
+{
+    char *curve = 0;
+    size_t namelen;
+
+    if (EVP_PKEY_get_group_name(pkey, NULL, 0, &namelen)) {
+        curve = OPENSSL_malloc(++namelen);
+        if (!EVP_PKEY_get_group_name(pkey, curve, namelen, 0)) {
+            OPENSSL_free(curve);
+            curve = NULL;
+        }
+    }
+    return (curve);
+}
+#endif
```

Comment:
IMO this does not need to be inside such `ifndef`. What if a 3rd party provider or the fips provider supports EC?

Reply:
Sure, I'll remove the guards.